src/elan/json.rs

```diff
@@ -0,0 +1,65 @@
+use Cfg;
+use std::path::PathBuf;
+use std::io::Result;
+
+use serde_derive::Serialize;
+
+use crate::OverrideReason;
+
+#[derive(Serialize)]
+struct Version {
+    current: String,
+    /// `None` on network error
+    newest: Option<String>,
+}
+
+#[derive(Serialize)]
+struct InstalledToolchain {
```

Comment:
How to figure out whether a call to `lean` will download a new Lean version with this format (e.g. if a toolchain is `stable`)?

Reply:
If the active, resolved toolchain is not in the installed toolchains, it will be downloaded. Let's put the active toolchain in there as well to make things easier.

Reply:
Ah, right. I forgot that resolution won't itself download the toolchain.